QueuePriorityTest.queueConsistency

http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/4290/

```
java.lang.AssertionError: expected:<1000> but was:<999>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.collection.impl.queue.QueuePriorityTest.queueConsistency(QueuePriorityTest.java:186)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueuePriorityTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueuePriorityTest.java
@@ -32,6 +32,8 @@ import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.concurrent.ConcurrentSkipListSet;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
@@ -165,11 +167,19 @@ public class QueuePriorityTest extends HazelcastTestSupport {
     @Test
     public void queueConsistency() {
         int count = 0;
-        for (int i = 0; i < 500; i++) {
+        int itemCount = 500;
+        List<PriorityElement> expected = new ArrayList<>(itemCount * 2);
+        for (int i = 0; i < itemCount; i++) {
             queue.offer(new PriorityElement(false, count));
             queue.offer(new PriorityElement(true, count));
+
+            expected.add(new PriorityElement(false, count));
+            expected.add(new PriorityElement(true, count));
             count++;
         }
+        // We don't expect total 1000 priority elements to fill the queue capacity
+        // and don't expect offers to fail
+        assertEquals(itemCount * 2, queue.size());
 
         ConcurrentSkipListSet<PriorityElement> tasks = new ConcurrentSkipListSet<>(new PriorityElementComparator());
         CountDownLatch latch = new CountDownLatch(100);
@@ -183,7 +193,7 @@ public class QueuePriorityTest extends HazelcastTestSupport {
             });
         }
         assertOpenEventually(latch);
-        assertEquals(500 * 2, tasks.size());
+        assertContainsAll(tasks, expected);
         assertNull(queue.poll());
     }
 

--- a/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/model/PriorityElement.java
+++ b/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/model/PriorityElement.java
@@ -62,4 +62,12 @@ public class PriorityElement implements Serializable {
     public int hashCode() {
         return Objects.hash(highPriority, version);
     }
+
+    @Override
+    public String toString() {
+        return "PriorityElement{" +
+                "highPriority=" + highPriority +
+                ", version=" + version +
+                '}';
+    }
 }

--- a/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/model/PriorityElement.java
+++ b/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/model/PriorityElement.java
@@ -65,9 +65,9 @@ public class PriorityElement implements Serializable {
 
     @Override
     public String toString() {
-        return "PriorityElement{" +
-                "highPriority=" + highPriority +
-                ", version=" + version +
-                '}';
+        return "PriorityElement{"
+                + "highPriority=" + highPriority
+                + ", version=" + version
+                + '}';
     }
 }

